woocommerce/woocommerce
# Add to Cart with Options: Add Stepper option

https://github.com/user-attachments/assets/15cf65ee-d66b-4d04-8f3d-701e6d60d15a

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Closes #42299.

This PR improves the Add to Cart with Options block:
- it moves the block from the Atomic folder to the Blocks folder (check #42470).
- It adds the Quantity Selector option to change the block's style.
- it improves the default style.

### It moves the block from the Atomic folder to the Blocks folder 

With #42470, we want to move all the product element blocks from the atomic folder. I did in this PR because it was necessary for tweaking the webpack configuration to add the Interactivity API support to this block.


### It adds the Quantity Selector option to change the block's style



This PR adds a new option to change the block's style. Also, when the `stepper` mode is enabled, the Interactivity API changes the input value when the user clicks on the `-` or `+` button.


## Screenshots

### Single Product Template

#### Input

|  Product Type  | TT4 | TT3 |
|--------|--------|--------|
| Simple Product |![image](https://github.com/user-attachments/assets/c48d8352-950a-4151-b035-ed1e4901b918)|![image](https://github.com/user-attachments/assets/37b8020b-70df-4487-b38d-d1bfd8e0496b)|
| Product Variation |![image](https://github.com/user-attachments/assets/d4092083-7ec2-4f6c-94a8-4193a20ceaab)| ![image](https://github.com/user-attachments/assets/28e1c003-ac5b-414a-8378-59694ecb54b0) |
| Grouped Product |![image](https://github.com/user-attachments/assets/add0e5bc-2754-4f69-abf2-9ab5890bd2e6)|![image](https://github.com/user-attachments/assets/816c21a4-0a46-41c0-9527-a27cecbee097) | 

#### Stepper

|  Product Type  | TT4 | TT3 |
|--------|--------|--------|
| Simple Product | ![image](https://github.com/user-attachments/assets/3424a205-46ce-400a-bca6-7f68b92df093)| ![image](https://github.com/user-attachments/assets/05cc0fb0-61ed-408b-870a-2a1f70db1c4b)) |
| Product Variation |![image](https://github.com/user-attachments/assets/d60ee621-176c-4985-9d77-65e799820485)| ![image](https://github.com/user-attachments/assets/d475dac5-9089-461b-bcc2-1190fc5582f8) |
| Grouped Product |![image](https://github.com/user-attachments/assets/6a91512a-8f0b-4dfd-ac37-8d115e26cd41) |![image](https://github.com/user-attachments/assets/2d1cf36d-c170-4a46-a22f-0f86405359b3)| 


### Single Product Block

Currently, the Single Product block has a bug and the style isn't loaded: https://github.com/woocommerce/woocommerce/issues/45369


### Compatibility and next steps

While efforts are made to ensure compatibility of this new mode with other extensions, it cannot be guaranteed. Certain themes/extensions that heavily customize CSS and button layouts might disrupt this new setting. Consequently, this mode will **not be** the default. Should this PR be merged, a development advisory or dedicated post will be published to inform theme/extension developers, allowing them to test compatibility with their products.






### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable TT4.
2. Install Woo Payments, WooCommerce PayPal Payments, WooCommerce Stripe Gateway and configure them ensuring that the express checkout button are visible.
3. Edit the Single Product Template.
4. Ensure that you have the Add To Cart Form block added.
5. Click on it.
6. Open the block settings.
7. Ensure that you have the `Quantity Selector` option
8. Ensure that the default one is `Input`.
9. Ensure that the input looks like this image:

![image](https://github.com/woocommerce/woocommerce/assets/4463174/a2263956-e591-4dc4-ac63-33695cb6d6ee)


10. Switch the option to `Stepper`
11. Ensure that the input looks like this image:
![image](https://github.com/woocommerce/woocommerce/assets/4463174/49585883-1798-4353-afb4-62083512d278)
**The following steps should be checked for both options and with a Simple Product, a Variation Product, and a Grouped Product**
12. Visit the Product.
13. Ensure that the input looks like the image in [the table ](https://github.com/woocommerce/woocommerce/pull/47664#issuecomment-2137043597).
14. Ensure that the input works correctly (pay attention to focus/border/etc...).
15. Click Add to Cart.
16. Ensure that the right amount of the products is added.
17. Repeat with TT3 or another block theme.
18. Create a new post.
**The following steps should be checked for both options and with a Simple Product, a Variation Product and a Grouped Product**
19. Add the `Single Product` block
11. Visit the page.
12. Ensure that the input looks like the image [in the table](https://github.com/woocommerce/woocommerce/pull/47664#issuecomment-2137043597).
13. Ensure that the input works correctly (pay attention to focus/border/etc...).
14. Click Add to Cart.
15. Ensure that the right amount of the products is added.
16. Repeat with TT3 or another block theme.
17. Testing this PR with Firefox and Chrome.

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add to Cart with Options: Add Quantity Selector option.



</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
